services/document-manager/backend/app/services/object_store_storage_service.py

```diff
@@ -249,4 +253,86 @@ def upload_zip_file(self, file_data, key, file_size, progress_callback):
             
             return True
         except ClientError as e:
-            raise Exception(f'Failed to upload the file: {e}')
\ No newline at end of file
+            raise Exception(f'Failed to upload the file: {e}')
+
+    def __download_in_chunks(self, key):
+        # Generator function to download file in chunks of 8MB size
+
+        # Get the object from S3
+        response = self._client.get_object(Bucket=Config.OBJECT_STORE_BUCKET, Key=key)
+
+        # Read the object in chunks and yield them one by one
+        for chunk in response['Body'].iter_chunks(chunk_size=8 * 1024 * 1024):
+            yield chunk
+
+    def create_zip_and_upload(self, key, document_paths, pretty_paths, progress_callback):
+        # Calculate the total size of all the files
+        total_size = sum(
+            self._client.head_object(Bucket=Config.OBJECT_STORE_BUCKET, Key=path)['ContentLength'] for path in
+            document_paths)
+
+        def local_files():
+            # Get the current timestamp
+            now = datetime.now()
+
+            def contents(path):
+                # Generator to read the file in chunks from S3
+                for chunk in self.__download_in_chunks(path):
+                    yield chunk
+
+            # Return a generator with tuples of file parameters for each file
+            return (
+                (pretty_path, now, S_IFREG | 0o600, ZIP_32, contents(document_path))
+                for document_path, pretty_path in zip(document_paths, pretty_paths)
+            )
+
+        def to_file_like_obj(iterable):
+            # Converts an iterator to a file-like object
+
+            chunk = b''
+            offset = 0
+            it = iter(iterable)
+
+            def up_to_iter(size):
+                # Internal function to yield chunks from the iterable
+
+                nonlocal chunk, offset
+                while size:
+                    if offset == len(chunk):
+                        try:
+                            chunk = next(it)
+                        except StopIteration:
+                            break
+                        else:
+                            offset = 0
+                    to_yield = min(size, len(chunk) - offset)
+                    offset = offset + to_yield
+                    size -= to_yield
+                    yield chunk[offset - to_yield:offset]
+
+            class FileLikeObj:
+                # File-like object to handle reading in chunks
```

Comment:
```suggestion
                # A class that creates an object which behaves like a file. Instead of reading from disk, we're reading
            from the generator.
```